app/src/components/submissionPage/SubmissionPage.tsx

```diff
@@ -15,16 +17,33 @@ const WelcomeMessage = styled.p`
 `;
 
 const SubmissionPage = (props: { snappables: MentionElements[] }) => {
+    const [status, setStatus] = useState<string>("Loading...");
+    const [cup, setCup] = useState<CupWithId | undefined>(undefined);
+
+    useEffect(() => {
+        setStatus("Loading...");
+        (async () => {
+            try {
+                setCup(await getCurrentCupIfExists());
+                setStatus("");
+            } catch (err) {
+                console.error("Getting cup", err);
+                setStatus("There was an unexpected error loading the snapcup.");
+            }
+        })();
+    }, [setCup]);
+
     return (
         <>
             <WelcomeMessage>
                 Welcome, {getCurrentUserName().split(" ")[0]!}
             </WelcomeMessage>
-            <SubmissionTextBox
+            <TextBoxIfSnapcupOpen
+                cup={cup}
                 snappables={props.snappables}
-                user={getCurrentUserName()}
+                status={status}
```

Comment:
The `SubmissionPage` should deal with the loading state, no need to pass it to the child.